config/nginx.conf

```diff
@@ -0,0 +1,60 @@
+server {
+    listen 80 default_server;
+    root /srv/omeka;
+
+    location @app {
+        rewrite ^(/.*)/$ $1;
+        set $front_controller "index.php";
+        fastcgi_pass unix:/run/php-fpm/www.sock;
+        fastcgi_split_path_info ^(.+\.php)(/.*)$;
+
+        include fastcgi_params;
+        fastcgi_param SCRIPT_FILENAME $realpath_root/$front_controller;
+        fastcgi_param DOCUMENT_ROOT $realpath_root;
+        fastcgi_param REQUEST_URI   $uri$is_args$args;
+
+        internal;
+    }
+
+    location ~ ^(/\.|config|application) {
+        deny all;
+        access_log off;
+        log_not_found off;
+    }
+
+    location ~ ^/sorting-room {
+        index  index.html;
+        if (-d $request_filename) {
+            rewrite [^/]$ $scheme://$http_host$uri/ permanent;
+        }
+    }
+
+    location ~ ^/presley/(.+) {
+        try_files $uri @app;
+    }
+
+    location ~ ^/s/(.+)/auth/token {
+        try_files $uri @app;
+    }
+
+    location ~ /api/iiif-import {
+        try_files $uri @app;
+    }
+
+    location / {
+        if ($request_method = OPTIONS ) {
+            add_header Access-Control-Allow-Origin "*";
+            add_header Access-Control-Allow-Methods "GET, POST, PUT, OPTIONS";
+            add_header Access-Control-Allow-Headers "Content-Type,Slug";
+            add_header Access-Control-Allow-Credentials "true";
+            add_header Content-Length 0;
+            add_header Content-Type text/plain;
+            return 200;
+        }
+
+        try_files $uri @app;
+    }
+
+    error_log /var/log/nginx/error.log;
+    access_log /var/log/nginx/access.log;
+}
```

Comment:
@garyttierney I was going to put this upstream into the Madoc docker image, but the locations are very specific to the modules in madoc. The base madoc image is really just madoc without any modules. Additions here compared to the base: added `presley` to the list of URLs that bypass the automatic `OPTIONS` check, since they support authorisation. There is also a static serving of the sorting room.